apps/prairielearn/elements/pl-matrix-component-input/pl-matrix-component-input.py

```diff
@@ -372,14 +397,20 @@ def grade(element_html: str, data: pl.QuestionData) -> None:
     weight = pl.get_integer_attrib(element, "weight", WEIGHT_DEFAULT)
 
     # Get method of comparison, with relabs as default
-    comparison = pl.get_string_attrib(element, "comparison", COMPARISON_DEFAULT)
-    if comparison == "relabs":
+    comparison = pl.get_enum_attrib(
+        element, "comparison", ComparisonMode, COMPARISON_DEFAULT
+    )
+
+    rtol, atol, digits = RTOL_DEFAULT, ATOL_DEFAULT, DIGITS_DEFAULT
+    if comparison == ComparisonMode.RELABS:
         rtol = pl.get_float_attrib(element, "rtol", RTOL_DEFAULT)
         atol = pl.get_float_attrib(element, "atol", ATOL_DEFAULT)
-    elif comparison in ("sigfig", "decdig"):
+    elif comparison in (ComparisonMode.SIGFIG, ComparisonMode.DECDIG):
         digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
     else:
-        raise ValueError(f'method of comparison "{comparison}" is not valid')
+        raise ValueError(
+            f'method of comparison "{comparison}" is not valid (must be "relabs", "sigfig", or "decdig")'
+        )
```

Comment:
Can we `assert_never(comparison)` here instead? enum parsing should already handle the case where this has some other value. Ditto elsewhere in this file.

Reply:
Changed in 097ef66c6cb47bc0f959188a4b0a2f4dd5a4a10c.